2_dkr.py

```diff
@@ -0,0 +1,309 @@
+        print("Current Hand:", end=" ")
+        display_hand(hand_)
+        inputings = input("Enter word, or “!!” to indicate that you are finished:")
+        if is_valid_word(inputings, hand_, word_list):
+            count_points += get_word_score(inputings, calculate_handlen(hand_))
+            print(f"With the word ' {inputings} 'you've got {get_word_score(inputings, calculate_handlen(hand_))}"
+                  f" points Scores:  {count_points} points)")
+            print(" ")
+        elif inputings == "!!":
+            break
+        else:
+            print("This is not valid word. Please choose another word", "\n")
+        hand_ = update_hand(hand_, inputings)
+    else:
+        print()
+        print("Ran out of letters.")
+    return count_points
+
+
+def substitute_hand(hand, letter):
+    """
+    Allow the user to replace all copies of one letter in the hand (chosen by user)
+    with a new letter chosen from the VOWELS and CONSONANTS at random. The new letter
+    should be different from user's choice, and should not be any of the letters
+    already in the hand.
+
+    If user provide a letter not in the hand, the hand should be the same.
+
+    Has no side effects: does not mutate hand.
+
+    For example:
+        substitute_hand({'h':1, 'e':1, 'l':2, 'o':1}, 'l')
+    might return:
+        {'h':1, 'e':1, 'o':1, 'x':2} -> if the new letter is 'x'
+    The new letter should not be 'h', 'e', 'l', or 'o' since those letters were
+    already in the hand.
+
+    hand: dictionary (string -> int)
+    letter: string
+    returns: dictionary (string -> int)
+    """
+    try:
+        letter_to_be_changed = hand[letter]
+    except:
+        return hand
+
+    is_true = True
+    while is_true:
+        new = random.choice(string.ascii_letters.lower())
+        if new not in hand:
+            del hand[letter]
+            hand.update({new: letter_to_be_changed})
+            is_true = False
+
+        continue
+
+    return hand
+
+
+def play_game(word_list):
+    """
+    Allow the user to play a series of hands
+    word_list: list of lowercase strings
+    """
+
+    all_points = 0
+    print("Let's start the game!")
+    number_of_hands = int(input("Input the total number of hands: "))
+    while number_of_hands != 0:
+        hand = deal_hand(HAND_SIZE)
+        print("Current hand: ", end=" ")
+        display_hand(hand)
+        answer = str(input("Would you like to substitute a letter?(yes/no) "))
+        if answer == "yes":
+            r = str(input("Choose the letter to be changed, please: \n"))
+            hand = substitute_hand(hand, r)
+        else:
+            print()
+        hand_points = play_hand(hand, word_list, 0)
+        all_points += hand_points
+        second_chance = input("Do you want to replay this hand?yes/no")
+        if second_chance == 'yes':
+            hand_points = play_hand(hand, word_list, 0)
+            all_points += hand_points
+            break

```

Comment:
Why do you make here `break`?
There can be more then one hand. So we replay the hand, we have just played.